src/libs/sys-util-rs/src/hooks.rs

```diff
@@ -0,0 +1,423 @@
+// Copyright (c) 2019-2021 Alibaba Cloud
+// Copyright (c) 2019-2021 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::collections::hash_map::DefaultHasher;
+use std::collections::HashMap;
+use std::ffi::OsString;
+use std::hash::{Hash, Hasher};
+use std::io::{self, Read, Result};
+use std::path::Path;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+
+use lazy_static::lazy_static;
+use subprocess::{ExitStatus, Popen, PopenConfig, PopenError, Redirection};
+
+use crate::{eother, sl};
+
+const DEFAULT_HOOK_TIMEOUT: i32 = 10;
```

Comment:
A comment giving the units would be helpful (I'm guessing seconds?).

Reply:
Or add the units to the `const` name to make the code self-documenting.

Reply:
renamed to "DEFAULT_HOOK_TIMEOUT_SEC".